File: 프로그래머스/1/42840. 모의고사/모의고사.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] answers) {
        int[] p1 = {1, 2, 3, 4, 5};
        int[] p2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] p3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        int[] score = new int[3];
        
        for(int i = 0; i < answers.Length; i++)
        {
            if(answers[i] == p1[i % p1.Length]) score[0]++;
            if(answers[i] == p2[i % p2.Length]) score[1]++;
            if(answers[i] == p3[i % p3.Length]) score[2]++;
        }
        
        int maxScore = score.Max();
        List<int> result = new List<int>();
        for(int i = 0; i < 3; i++)
        {
            if(maxScore == score[i]) result.Add(i + 1);
        }
        score = result.ToArray();
        return score;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int mx = -1000000;
    int a[5] = {1, 2, 3, 4, 5};
    int b[8] = {2, 1, 2, 3, 2, 4, 2, 5};
    int c[10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    int cnt1 = 0;
    int cnt2 = 0;
    int cnt3 = 0;
    
    for(int i = 0; i < answers.size(); i++) {
        if(answers[i] == a[i % 5]) cnt1++;
        if(answers[i] == b[i % 8]) cnt2++;
        if(answers[i] == c[i % 10]) cnt3++;
    }
    mx = max(max(cnt3, cnt2), cnt1);
    
    if(cnt1 == mx) answer.push_back(1);
    if(cnt2 == mx) answer.push_back(2);
    if(cnt3 == mx) answer.push_back(3);

    return answer;
}
// 1번 : 1 2 3 4 5
// 2번 : 2 1 2 3 2 4 2 5 
// 3번 : 3 3 1 1 2 2 4 4 5 5